main-app/app/scripts/providers/shuffle-and-deal.js

```diff
@@ -0,0 +1,23 @@
+(function () {
+    'use strict';
+    angular.module('Tombola.SnapCardGame.Deck')
+        .service('ShuffleAndDeal', function () {
+            var numberOfHands = 2,
+                hand1 = [],
+                hand2 = [];
+            this.shuffle = function (deck) {
+                var j, k, i = deck.length;
+                while (--i) {
+                    j = Math.floor(Math.random() * i);
+                    k = deck[i];
+                    deck[i] = deck[j];
+                    deck[j] = k;
+                }
+                return deck;
+            };
+            this.deal = function (deck) {
```

Comment:
The deal function should take an array (the deck) and the number of hands you want to deal to (the controller will hard-code into 2). 
One upshot of this is that you shoudld end up with an Array of hands - the fact that you have 'hand1'  'hand2' variables is a classic indicator in JS that an array is favoured.